### PR TITLE
Add Fedcurv example using TaskRunner API

### PR DESCRIPTION
**Pull request by ynonflumintel**

### CODE REVIEW

File: openfl/federated/task/fl_model.py [hunk 1]
```diff
@@ -75,10 +76,17 @@ def __init__(self, build_model, optimizer=None, loss_fn=None, **kwargs):
             self.runner.validate = lambda *args, **kwargs: build_model.validate(
                 self.runner, *args, **kwargs
             )
+        # TODO: not used anywhere - should it be removed?
```
Comment (teoparvanov) [inline openfl/federated/task/fl_model.py]: It's not very common indeed, but I saw one instance [here](https://github.com/securefederatedai/openfl/blob/develop/openfl-workspace/torch_llm_horovod/src/InHorovodLLMTrainer.py#L200).

File: openfl/federated/task/fl_model.py [hunk 2]
```diff
@@ -75,10 +76,17 @@ def __init__(self, build_model, optimizer=None, loss_fn=None, **kwargs):
...
         if hasattr(self.model, "train_epoch"):
             self.runner.train_epoch = lambda *args, **kwargs: build_model.train_epoch(
                 self.runner, *args, **kwargs
             )
+
+        # Used to hook the training function when debugging locally
+        if hasattr(self.model, "train_"):
+            self.runner.train_ = lambda *args, **kwargs: build_model.train_(
+                self.runner, *args, **kwargs
+            )
```
Comment (teoparvanov) [inline openfl/federated/task/fl_model.py:90]: is it just for local debugging?

**Review by teoparvanov — APPROVED** [review]
LGTM!

File: openfl/federated/task/fl_model.py [hunk 3]
```diff
@@ -75,10 +76,17 @@ def __init__(self, build_model, optimizer=None, loss_fn=None, **kwargs):
             self.runner.validate = lambda *args, **kwargs: build_model.validate(
                 self.runner, *args, **kwargs
             )
+        # TODO: not used anywhere - should it be removed?
         if hasattr(self.model, "train_epoch"):
             self.runner.train_epoch = lambda *args, **kwargs: build_model.train_epoch(
                 self.runner, *args, **kwargs
             )
+
+        # Used to hook the training function when debugging locally
```
Comment (kminhta) [inline openfl/federated/task/fl_model.py:86]: Is this change related to the FedCurv task runner example?

File: openfl/native/native.py [hunk 4]
```diff
@@ -215,7 +215,7 @@ def init(
     Args:
         workspace_template (str): The template that should be used as the
             basis for the experiment.  Defaults to 'default'.
-            Other options include are any of the template names
+            Other options include any of the template names
```
Comment (kminhta) [inline openfl/native/native.py]: I think we can omit this from this PR. Also because native API will be deprecated and removed eventually anyway

**Review by kminhta — CHANGES_REQUESTED** [review]
Thanks @ynonflumintel ! Looks good overall

**Request:** Can you include a README with commands to run this workspace as well as some information about FedCurv (especially highlighting the diff from the base `torch_cnn_histology`? Good to cite it as well: https://arxiv.org/pdf/1910.07796

It does not need to be a super in depth doc, but should provide users the ability to run it end to end

Also, a couple minor comments/questions below

File: openfl/federated/task/fl_model.py (hunk 3 above)
Comment (ynonflumintel) [reply openfl/federated/task/fl_model.py:86]: I needed it to debug fedcurv, I don't mind omitting it if you don't think it should be included in this PR

File: openfl/federated/task/fl_model.py (hunk 2 above)
Comment (ynonflumintel) [reply openfl/federated/task/fl_model.py:90]: When using a distributed deployment you can change the train function directly , so yes

File: openfl/federated/task/fl_model.py (hunk 3 above)
Comment (kminhta) [reply openfl/federated/task/fl_model.py:86]: Ohh, I think it's okay to keep then. In fact, probably an overdue update since most of new runners (definitely the case for torch) use `train_` instead of `train_epoch`

**Review by kminhta — APPROVED** [review]
LGTM! Thanks @ynonflumintel
